src/kdf.c

```diff
@@ -195,125 +510,155 @@ static int p11prov_hkdf_set_ctx_params(void *ctx, const OSSL_PARAM params[])
             return ret;
         }
 
-        rv = p11prov_digest_get_by_name(digest,
-                                        &hkdfctx->params.prfHashMechanism);
+        rv = p11prov_digest_get_by_name(digest, &hkdfctx->hash_mech);
         if (rv != CKR_OK) {
             ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_DIGEST);
             return RET_OSSL_ERR;
         }
-        P11PROV_debug("set digest to %lu", hkdfctx->params.prfHashMechanism);
+        P11PROV_debug("set digest to %lu", hkdfctx->hash_mech);
     }
 
     p = OSSL_PARAM_locate_const(params, OSSL_KDF_PARAM_MODE);
     if (p) {
-        int mode;
         if (p->data_type == OSSL_PARAM_UTF8_STRING) {
             if (OPENSSL_strcasecmp(p->data, "EXTRACT_AND_EXPAND") == 0) {
-                mode = EVP_KDF_HKDF_MODE_EXTRACT_AND_EXPAND;
+                hkdfctx->mode = EVP_KDF_HKDF_MODE_EXTRACT_AND_EXPAND;
             } else if (OPENSSL_strcasecmp(p->data, "EXTRACT_ONLY") == 0) {
-                mode = EVP_KDF_HKDF_MODE_EXTRACT_ONLY;
+                hkdfctx->mode = EVP_KDF_HKDF_MODE_EXTRACT_ONLY;
             } else if (OPENSSL_strcasecmp(p->data, "EXPAND_ONLY") == 0) {
-                mode = EVP_KDF_HKDF_MODE_EXPAND_ONLY;
+                hkdfctx->mode = EVP_KDF_HKDF_MODE_EXPAND_ONLY;
             } else {
-                mode = 1;
+                hkdfctx->mode = 1;
```

Comment:
Style: a named constant should be used?

Reply:
no, although a preexisting thing, this looks like a bug and should probably error out, as it means the string is unrecognized.

Reply:
Resolved in a fixup